### PR TITLE
    [runtime] Replace the C and perl versions of the genmdesc tool wi…

### PR DESCRIPTION
**Pull request by vargaz**
…th a python version.

    This has the following advantages:
    * There is only one implementation instead of two
    * The C version no longer needs to be compiled/recompiled.

    The disadvantage is the additional build time dependency on python.

The windows build will keep using the C version for now.

### CODE REVIEW

**Comment by vargaz** [comment]
This is a simplified version of:
https://github.com/mono/mono/pull/5163
it doesn't touch the windows build which will be fixed later.

File: mono/mini/Makefile.am.in [hunk 1]
```diff
@@ -702,49 +696,36 @@ MemoryIntrinsics.dll: MemoryIntrinsics.il
...
-if CROSS_COMPILING
-GENMDESC_PRG=perl $(srcdir)/genmdesc.pl $(arch_define) $(srcdir) $(GENMDESC_OPTS)
-else
-if WASM
-GENMDESC_PRG=perl $(srcdir)/genmdesc.pl $(arch_define) $(srcdir) $(GENMDESC_OPTS)
-else
-GENMDESC_PRG=./genmdesc $(GENMDESC_OPTS)
-endif
-endif !CROSS_COMPILING
+GENMDESC_PRG=python $(srcdir)/genmdesc.py $(target_define) $(srcdir)
```
Comment (alexanderkyte) [inline mono/mini/Makefile.am.in]: The command `python` can mean a recent python3 or a recent python2 depending on the linux distro / way that python is compiled. Could you please specify which interpreter you're trying to use?

Comment (vargaz) [reply mono/mini/Makefile.am.in]: It shouldn't matter.

Comment (lewurm) [reply mono/mini/Makefile.am.in]: that was already part of the discussion of the previous PR, alas it looks like some context got lost 😕  the script runs fine on both, python2 and 3.

**Pull request merged**